src/main/scala/amba/axi4/Parameters.scala

```diff
@@ -210,11 +210,13 @@ class AXI4IdMap(axi4: AXI4MasterPortParameters) extends IdMap[AXI4IdMapEntry] {
   private val sorted = axi4.masters.sortBy(_.id)
 
   val mapping: Seq[AXI4IdMapEntry] = sorted.map { case c =>
-    AXI4IdMapEntry(c.id, c.name)
+    // to conservatively state max number of transactions, assume every id has up to c.maxFlight and reuses ids between AW and AR channels
+    val maxTransactionsInFlight = c.maxFlight.map(_ * c.id.size * 2)
+    AXI4IdMapEntry(c.id, c.name, maxTransactionsInFlight)
   }
 }
 
-case class AXI4IdMapEntry(axi4Id: IdRange, name: String) extends IdMapEntry {
+case class AXI4IdMapEntry(axi4Id: IdRange, name: String, maxTransactionsInFlight: Option[Int]) extends IdMapEntry {
```

Comment:
For backwards compatibility, should this default to `None`? Or is case class this only used in this file, and if so, then should this case class be marked as private/protected?

Reply:
It's plausible it should be private, but I decided to just make it take default None for now